src/sctools/bam.py

```diff
@@ -213,33 +213,37 @@ def _cleanup(files_to_counts, files_to_names, rm_all=False):
     return list(files_to_names.values())
 
 
-def iter_tag_groups(tag: str, bam_iterator: Iterator[pysam.AlignedSegment]) -> Generator:
+def iter_tag_groups(
+        tag: str,
+        bam_iterator: Iterator[pysam.AlignedSegment],
+        filter_null: bool=False) -> Generator:
+
     # get first read and tag set
     reads = []
-    while not reads:
-
-        try:
-            read = next(bam_iterator)
-            current_tag = read.get_tag(tag)
-            reads.append(read)
-            break
-        except KeyError:  # todo this needs to be tracked so we know how many reads were invalid
-            continue
-        except StopIteration:
-            raise ValueError('No reads contained tag %s' % tag)
+    read = next(bam_iterator)
+    try:
+        current_tag = read.get_tag(tag)
+    except KeyError:
+        current_tag = None  # null tag is a category that gets emitted
```

Comment:
I think we should be checking for `filter_null` here in these `except` blocks and either `continue`ing in the bottom case or throwing an error here if `true`.  This way we dont even have to the least amount of work if we're filtering them out anyways.

Reply:
See below for response.